src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -140,11 +145,16 @@ private async Task<bool> ExecuteAsync(Common.ILogger log)
 
             EmbedInBinlog = GetFilesToEmbedInBinlog(dgFile);
 
+            if (RestoreNoCache)
+            {
+                log.LogWarning(Strings.Log_RestoreNoCacheInformation);
```

Comment:
Is this going to cause the message to be output twice in some scenarios because we are outputting it once in Program.cs and once in RestoreTask? I haven't tested this so I'm not sure if this will, just wanting to understand why we log this warning once here and once when parsing the command.

Reply:
I think it will not be outputted twice. Program.cs is for nuget.exe. And the RestoreNoCache check is for the situation when a user uses msbuild and passes the property RestoreNoCache set to true.